From 67f322d97344ffcbada94cb3f4dda7dbe17f5195 Mon Sep 17 00:00:00 2001
From: Mathis MARION <mathis.marion@grenoble-inp.org>
Date: Thu, 29 Jul 2021 16:27:10 +0200
Subject: [PATCH 08/18] [fmc-sw] Fix double ID release

The ida_simple_remove() was happeing twice: one in the
fmc_slot_release() function triggered by device_unregister(), and the
other in the fmc_carrier_del_slot()
---
 patches/fmc-sw.diff | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/patches/fmc-sw.diff b/patches/fmc-sw.diff
index e80a869..6e6539a 100644
--- a/patches/fmc-sw.diff
+++ b/patches/fmc-sw.diff
@@ -1,3 +1,23 @@
+diff --git a/drivers/fmc/fmc-core.c b/drivers/fmc/fmc-core.c
+index d2118e0..61fff4f 100644
+--- a/drivers/fmc/fmc-core.c
++++ b/drivers/fmc/fmc-core.c
+@@ -219,15 +219,11 @@ err_ida:
+  */
+ static void fmc_carrier_del_slot(struct fmc_slot *slot)
+ {
+-	int id;
+-
+ 	if (IS_ERR_OR_NULL(slot))
+ 		return;
+ 
+-	id = slot->dev.id;
+ 	fmc_slot_eeprom_del(slot);
+ 	device_unregister(&slot->dev);
+-	ida_simple_remove(&fmc_slot_ida, id);
+ }
+ 
+ /**
 diff --git a/drivers/fmc/fmc-eeprom.c b/drivers/fmc/fmc-eeprom.c
 index a000a0d..c3aa905 100644
 --- a/drivers/fmc/fmc-eeprom.c
-- 
2.27.0

